### ISSUE TITLE
Warning if merging observables and make this work with DefineMap.set ~8

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-stache-bindings#53]
From: https://github.com/canjs/canjs/issues/2314

If a parent to child binding is setup like:

``` js
<my-component {^address}="person.address"/>
```

With the rendered data like:

``` js
{person: new can.Map()}
```

If `<my-component>`'s `address` is set to another map,  this line:
https://github.com/canjs/can-stache-bindings/blob/master/can-stache-bindings.js#L699

Will end up calling something like:

```
person.attr( anotherMap, true)
```

This causes obvious problems.

We should warn that this is a problem. We also might want to change `.attr` and DefineMap's `.set` to serialize any map type object they get passed in.

**Comment by bmomberger-bitovi** [comment]
In the current test suite, the only path that triggers the code in question is doing a can-value binding between a multiple select and a can.List().  The example given above is suspect, because one-way bindings never reach the `.attr(obj, true)` path because it requires `mustBeACompute` to be true in `getComputeFrom.scope()`. That latter part is entirely dependent on the binding being two-way.

Assuming that the binding is two-way, then, it comes down to how the expression reads out of `parentExpression.value()` -- it seems like unless it's a List, it's always a compute.  Still investigating, but this may not be an issue anymore due to seemingly unrelated changes.

**Comment by bmomberger-bitovi** [comment]
After a little bit more diving, these are the conditions under which the code in question will fire:

* The binding is two-way
* The parent binding is a scope lookup
* The parent context is a plain object
* No other reads depend on this binding.

 e.g.  `<my-component {(foo)}="bar"/>` rendered with `{ bar: new CanMap() }`

Worth noting here that the parent binding's `bar` property cannot be swapped out, because the compute in the binding is registered to the original value and will not change over the lifetime of the binding.

**Comment by bmomberger-bitovi** [comment]
Closed via #129
